tls: Add descriptions for x509v3 alt name verification (tls.verify_hostname) (#1393)
fluent/fluent-bit-docs: administration/transport-security.md

| :--- | :--- | :--- |
| tls | enable or disable TLS support | Off |
| tls.verify | force certificate validation | On |

> | tls.verify\_hostname | force hostname validation for certificates | Off |

Reviewer: Slightly different wording to below.

    tls.ca_file /etc/certs/fluent.crt
    tls.vhost   fluent.example.com
```

### Verify subjectAltName

> When extracting the X509v3 Subject Alternative Name from a certificate:

Reviewer: I maybe would reword this a bit, using `Then` as well is not ideal.

Something like?
```suggestion.
By default, TLS verification of hostnames is not done automatically.
As an example, we can extract the X509v3 Subject Alternative Name from a certificate:
```

```
X509v3 Subject Alternative Name:
    DNS:my.fluent-aggregator.net
```

> Then, the TLS client will connect to `other.fluent-aggregator.net`, there is a glitch of hostnames.

Reviewer: ```suggestion
As you can see, this certificate covers only `my.fluent-aggregator.net` so if we use a different hostname it should fail.
```

> This difference of the hostname can be detected with:

Reviewer: ```suggestion
To fully verify the alternative name and demonstrate the failure we enable `tls.verify_hostname`:
```
